pennylane/devices/null_qubit.py

```diff
@@ -277,6 +277,8 @@ def __init__(self, wires=None, shots=None, track_resources=False) -> None:
         self._debugger = None
         self._track_resources = track_resources
 
+        self.device_kwargs = {"track_resources": track_resources}
+
```

Comment:
Nice! A code comment would be nice here :+1: 

something like,
```suggestion

        # this is required by Catalyst to toggle the tracker at runtime
        self.device_kwargs = {"track_resources": track_resources}

```